src/main/resources/assets/react/documentTypes/NewOrEditDocumentType.jsx

```diff
@@ -104,8 +107,42 @@ export function NewOrEditDocumentType({
 			});
 	}
 
+	/**
+	 * Gets all the names of existing documentTypes
+	 *
+	 */
+	function getDocumentTypeNames() {
+		fetch(`${servicesBaseUrl}/graphQL`, {
+			method: 'POST',
+			headers: {
+				'Content-Type':	'application/json'
+			},
+			body: JSON.stringify({
+				query: GQL_QUERY_DOCUMENT_TYPE_NAMES
+			})
+		})
+			.then(response => response.json())
+			.then(json => json.data)
+			.then(data => {
+				const allNames = data.queryDocumentTypes.hits.map(({_name}) => _name);
+				setNames(() => allNames);
+			});
+	}
+
+	React.useEffect(() =>{
+		if (names.includes(nameInput)) {
```

Comment:
Since fetch is async, names will be an empty list the first time this code is run